### ISSUE TITLE
Slow nuxt startup when importing sass file

### ISSUE THREAD

**Issue by dikesh** [nuxt/framework#4613]
### Environment

- Operating System: `Linux`
- Node Version:     `v16.14.0`
- Nuxt Version:     `3.0.0-rc.1`
- Package Manager:  `npm@8.3.1`
- Builder:          `vite`
- User Config:      `css`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-cimcyj

### Describe the bug

I am using Bulma CSS with Nuxt app.

When I `@import 'node_modules/bulma/bulma.sass';` in my `'@/assets/main.scss'` file. the nuxt startup becomes too slow. It takes around 20-60 seconds for vite client and server to warm up.

![image](https://user-images.githubusercontent.com/1618526/165240771-bb4697a7-02d8-4d97-8924-bd660238e648.png)

Currently I have app running with `nuxt3@3.0.0-27460489.53fbca7` without this issue.

**Comment by dikesh** [comment]
@danielroe Not sure whether the issue is resolved or not but the start-up time has reduced in RC2

![image](https://user-images.githubusercontent.com/1618526/166624474-22012d69-e332-455b-a7a3-cf1593aaaf79.png)

https://stackblitz.com/edit/nuxt-starter-gt6w76

**Comment by dan-hammond** [comment]
I just wanted to add that I think this is still an issue in RC3. I have also noticed a few others reporting the same issue in the Nuxt Discord.

**Comment by dikesh** [comment]
Closing the issue as haven't faced this problem since v3.0.0-rc.2

**Comment by michastubi** [comment]
Is this problem really solved? I still facing this issue with version 3.0.0

Vite client warmed up in 43303ms
Nitro built in 3393 ms

If I remove /assets/main.scss from nuxt.config.ts, the startup time goes down to just two seconds.
Is this normal?

Here is the link to stackblitz using nuxt 3.0.0
[https://stackblitz.com/edit/nuxt-starter-edue27?file=nuxt.config.ts](https://stackblitz.com/edit/nuxt-starter-edue27?file=nuxt.config.ts)

**Comment by dikesh** [comment]
@michastubi Yes I am facing the issue again.
Vite client is taking longer to warn up.

![image](https://user-images.githubusercontent.com/1618526/204290350-a53feae7-ae23-4b80-a4f2-54549247e414.png)

**Comment by ayalon** [comment]
I have the same issue with including Bulma.
35s to start up as soon as I include a CSS file.

**Comment by dulnan** [comment]
## tl;dr Replace custom `divide` function with `math.div`.

After some longer profiling and debugging I think I found out the **actual reason** for very slow warm ups and builds (mine took anywhere from 20 seconds to 2 minutes). At least in my case it reduced the build time significantly.

Both Bulma and Bootstrap have a magic `divide` function that reimplements... dividing numbers. Originally this was implemented in Bootstrap and Bulma copied that code from there, but Bootstrap has since then "updated" the function to be more "efficient":

https://github.com/twbs/bootstrap/blob/main/scss/_functions.scss#L262

Here's Bulma's code (based on an older version from Bootstrap):
https://github.com/jgthms/bulma/blob/master/sass/utilities/functions.sass#L118

The main issue seems to be a while loop, looping itself into oblivion. After replacing it with `math.div()` from `sass:math`, the warm up time dropped down to **4 seconds** from 22 seconds. I have also tried the *improved* version from Bootstrap, which reduced the same build time to 16 seconds.

Now calling this function just a few dozen times in the beginning might not be an issue, but if you have dozens of SCSS files, components with styles _and_ you're using `additionalData` to provide global variables, you'll end up running that inefficient code hundreds of times. Which adds up to a lot, hence why builds take ages.

**Comment by danielroe** [comment]
Really appreciate your time diving into this. Thank you.
